rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -1555,6 +1562,26 @@ void getPendingFederationHash_returnsPendingFederationHash() {
             Keccak256 pendingFederationHash = federationSupport.getPendingFederationHash();
             assertThat(pendingFederationHash, is(pendingFederation.getHash()));
         }
+
+        @Test
+        @Tag("getPendingFederatorBtcPublicKey")
+        void getPendingFederatorBtcPublicKey_returnsFederatorFromPendingFederationBtcPublicKey() {
+            byte[] pendingFederatorBtcPublicKey = federationSupport.getPendingFederatorBtcPublicKey(0);
+            assertThat(pendingFederatorBtcPublicKey, is(pendingFederation.getBtcPublicKeys().get(0).getPubKey()));
+        }
+
+        @Test
+        @Tag("getPendingFederatorBtcPublicKey")
+        void getPendingFederatorBtcPublicKey_withNegativeIndex_throwsIndexOutOfBoundsException() {
+            assertThrows(IndexOutOfBoundsException.class, () -> federationSupport.getPendingFederatorBtcPublicKey(-1));
```

Comment:
I am wandering, are we expecting this to happen in prod? Does it make sense to include this test? If that is the case we should probably be catching this exception inside `getPendingFederatorBtcPublicKey` and throwing a more domain custom one. Just an idea not sure.

Reply:
I see, that exception should already be being thrown by the `get` method to the list. Not sure why that logic is there.

Reply:
Good catch @apancorb, it's the very same exact exception.

We could simply remove that piece of code, or catch the exception and throw a custom one. Thoughts? @julia-zack @apancorb

Reply:
ahhh you right guys. 
i don't see any benefits from catching the exception, its just an index out of bounds exception.
I think I would simply remove the piece of code

Reply:
If this is not expected to happen in prod then I would just remove that logic. Otherwise, catch and throw a custom one.

Reply:
Apparently this method can only be called before RSKIP123, it was then replaced for `getPendingFederatorPublicKeyOfType`.

So I simply removed the check